src/code/code_800EC960.c

```diff
@@ -824,354 +833,395 @@ NatureAmbienceDataIO sNatureAmbienceDataIO[20] = {
+        { OCARINA_PITCH_NONE, 0, 255, 0, 0, 0 },
     },
 
-    // Lost Woods Memory Game
+    // 13: Lost Woods Memory Game
     {
-        { 2, 0, 3, 0, 0, 0, 0 },
-        { 0xFF, 0, 0, 0, 0, 0, 0 },
+        { OCARINA_PITCH_D4, 3, 0, 0, 0, 0 },
+        { OCARINA_PITCH_NONE, 0, 0, 0, 0, 0 },
     },
 };
 
-OcarinaNote* sPlaybackSong = sOcarinaSongs[0];
+OcarinaNote* sPlaybackSong = sOcarinaSongNotes[0];
 u8 sFrogsSongNotes[14] = {
-    OCARINA_NOTE_A,       OCARINA_NOTE_C_LEFT,  OCARINA_NOTE_C_RIGHT, OCARINA_NOTE_C_DOWN, OCARINA_NOTE_C_LEFT,
-    OCARINA_NOTE_C_RIGHT, OCARINA_NOTE_C_DOWN,  OCARINA_NOTE_A,       OCARINA_NOTE_C_DOWN, OCARINA_NOTE_A,
-    OCARINA_NOTE_C_DOWN,  OCARINA_NOTE_C_RIGHT, OCARINA_NOTE_C_LEFT,  OCARINA_NOTE_A,
+    OCARINA_BTN_A,       OCARINA_BTN_C_LEFT,  OCARINA_BTN_C_RIGHT, OCARINA_BTN_C_DOWN, OCARINA_BTN_C_LEFT,
+    OCARINA_BTN_C_RIGHT, OCARINA_BTN_C_DOWN,  OCARINA_BTN_A,       OCARINA_BTN_C_DOWN, OCARINA_BTN_A,
+    OCARINA_BTN_C_DOWN,  OCARINA_BTN_C_RIGHT, OCARINA_BTN_C_LEFT,  OCARINA_BTN_A,
 };
 u8* gFrogsSongPtr = sFrogsSongNotes;
-u8 sRecordingState = 0;
+u8 sRecordingState = OCARINA_RECORD_OFF;
 u8 sRecordSongPos = 0;
-u32 D_80131860 = 0;
-u8 D_80131864 = 0;
-u8 D_80131868 = 0;
-u8 D_8013186C = 0;
-s8 D_80131870 = 0;
-u8 D_80131874 = 0;
-u8 D_80131878 = 0;
-u8 D_8013187C = 0;
-u8 D_80131880 = 0;
-
-OcarinaNote sPierresSong[108] = {
-    { 0xFF, 0, 0, 0, 0, 0, 0 },
-    { 0xFF, 0, 0, 0, 0, 0, 0 },
+u32 sOcarinaRecordTaskStart = 0;
+u8 sRecordOcarinaPitch = 0;
+u8 sRecordOcarinaVolume = 0;
+u8 sRecordOcarinaVibrato = 0;
+s8 sRecordOcarinaBendIdx = 0;
+u8 sRecordOcarinaButtonIdx = 0;
+u8 sPlayedOcarinaSongIdxPlusOne = 0;
+u8 sMusicStaffNumNotesPerTest = 0;
+u8 sIsOcarinaNoteChanged = false;
+
+OcarinaNote sScarecrowsLongSongNotes[108] = {
+    { OCARINA_PITCH_NONE, 0, 0, 0, 0, 0 },
+    { OCARINA_PITCH_NONE, 0, 0, 0, 0, 0 },
+};
+OcarinaNote* gScarecrowLongSongPtr = sScarecrowsLongSongNotes;
+
+u8* gScarecrowSpawnSongPtr = (u8*)&sOcarinaSongNotes[OCARINA_SONG_SCARECROW_SPAWN];
+OcarinaNote* sMemoryGameSongPtr = sOcarinaSongNotes[OCARINA_SONG_MEMORY_GAME];
+
+u8 sPitchToButtonMap[16] = {
+    OCARINA_BTN_A,                            // OCARINA_PITCH_C4
+    OCARINA_BTN_A,                            // OCARINA_PITCH_DFLAT4
+    OCARINA_BTN_A,                            // OCARINA_PITCH_D4
+    OCARINA_BTN_A,                            // OCARINA_PITCH_EFLAT4
+    OCARINA_BTN_C_DOWN,                       // OCARINA_PITCH_E4
+    OCARINA_BTN_C_DOWN,                       // OCARINA_PITCH_F4
+    OCARINA_BTN_C_DOWN,                       // OCARINA_PITCH_GFLAT4
+    OCARINA_BTN_C_RIGHT,                      // OCARINA_PITCH_G4
+    OCARINA_BTN_C_RIGHT,                      // OCARINA_PITCH_AFLAT4
+    OCARINA_BTN_C_RIGHT,                      // OCARINA_PITCH_A4
+    OCARINA_BTN_C_RIGHT + OCARINA_BTN_C_LEFT, // OCARINA_PITCH_BFLAT4: Interface/Overlap between C_RIGHT and C_LEFT
```

Comment:
This was quite the interesting case. I could see it being interpreted either way, it's hard to say what the intent was. I put it as a sum because introducing a new enum for `5` seemed like that enum value wouldn't follow what the rest of the enum was doing. `OCARINA_BTN_C_RIGHT_AND_C_LEFT` isn't really a button and that could be confusing since you can't push down both C_RIGHT and C_LEFT at the same time. And the `AND` and `+` could be loosely interchangeable in certain mathematical contexts. 

The actual numerical number 5 or the sum of the two enums isn't significant, it's to show that it's a hybrid between the two and a special case. I'm not inherently opposed to introducing `OCARINA_BTN_C_RIGHT_AND_C_LEFT` or an alt name, but I like the sum since it's in the middle of the two. Just thought I'd share my motivation.

Reply:
if the source was to never change I'd be fine with it, but looking at it in a context where the source is supposed to be as shiftable as can be, changing the enum like
```diff
 typedef enum {
+    OCARINA_BTN_C_RIGHT,
+    OCARINA_BTN_C_LEFT,
     OCARINA_BTN_A,
     OCARINA_BTN_C_DOWN,
-    OCARINA_BTN_C_RIGHT,
-    OCARINA_BTN_C_LEFT,
     OCARINA_BTN_C_UP,
     /* -1 */ OCARINA_BTN_INVALID = 0xFF
 } OcarinaButtonIdx;
```
afaik wouldn't break anything (?), *except* c_right + c_left would now sum to c_left instead of a whole new value

so I think I'd want a `OCARINA_BTN_C_RIGHT_AND_C_LEFT` value that shifts appropriately, if it makes sense

Reply:
Hmmm yeah that's a fair point about shiftability. I might use `OCARINA_BTN_C_RIGHT_OR_C_LEFT` since it's not pressing both C-Right AND C-Left, it's resolving to either C-Right OR C-Left and I'll add a comment explaining that difference